.gitignore

```diff
@@ -1,32 +1,489 @@
-*.class
+########################
+### project-specific ###
+########################
+# Add your ignores here
 
-# Log file
-*.log
+######################
+### auto-generated ###
+######################
+# Created by https://www.toptal.com/developers/gitignore/api/windows,linux,macos,vim,emacs,jetbrains,visualstudiocode,sublimetext,rust,dotenv,react,node
+# Edit at https://www.toptal.com/developers/gitignore?templates=windows,linux,macos,vim,emacs,jetbrains,visualstudiocode,sublimetext,rust,dotenv,react,node
+
+### dotenv ###
+.env
+
+### Emacs ###
+# -*- mode: gitignore; -*-
+*~
+\#*\#
+/.emacs.desktop
+/.emacs.desktop.lock
+*.elc
+auto-save-list
+tramp
+.\#*
+
+# Org-mode
+.org-id-locations
+*_archive
+
+# flymake-mode
+*_flymake.*
+
+# eshell files
+/eshell/history
+/eshell/lastdir
+
+# elpa packages
+/elpa/
+
+# reftex files
+*.rel
+
+# AUCTeX auto folder
+/auto/
+
+# cask packages
+.cask/
+dist/
+
+# Flycheck
+flycheck_*.el
+
+# server auth directory
+/server/
+
+# projectiles files
+.projectile
+
+# directory configuration
+.dir-locals.el
+
+# network security
+/network-security.data
+
+
+### JetBrains ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+
+# AWS User-specific
+.idea/**/aws.xml
+
+# Generated files
+.idea/**/contentModel.xml
+
+# Sensitive or high-churn files
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+.idea/**/dbnavigator.xml
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Gradle and Maven with auto-import
+# When using Gradle or Maven with auto-import, you should exclude module files,
+# since they will be recreated, and may cause churn.  Uncomment if using
+# auto-import.
+# .idea/artifacts
+# .idea/compiler.xml
+# .idea/jarRepositories.xml
+# .idea/modules.xml
+# .idea/*.iml
+# .idea/modules
+# *.iml
+# *.ipr
+
+# CMake
+cmake-build-*/
+
+# Mongo Explorer plugin
+.idea/**/mongoSettings.xml
+
+# File-based project format
+*.iws
+
+# IntelliJ
+out/
+
+# mpeltonen/sbt-idea plugin
+.idea_modules/
+
+# JIRA plugin
+atlassian-ide-plugin.xml
 
-# BlueJ files
-*.ctxt
+# Cursive Clojure plugin
+.idea/replstate.xml
 
-# Mobile Tools for Java (J2ME)
-.mtj.tmp/
+# SonarLint plugin
+.idea/sonarlint/
 
-# Package Files #
-*.jar
-*.war
-*.nar
-*.ear
-*.zip
-*.tar.gz
-*.rar
+# Crashlytics plugin (for Android Studio and IntelliJ)
+com_crashlytics_export_strings.xml
+crashlytics.properties
+crashlytics-build.properties
+fabric.properties
 
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+# Editor-based Rest Client
+.idea/httpRequests
 
-# folders
-org
-pdfs
-target
```

Comment:
Please don't remove these folders, I want them to be ignored.

Reply:
readded the first two with https://github.com/ElektraInitiative/PermaplanT/pull/113/commits/bd1f7916657fb4f3a39196beeab759cf8c5d4091.
`target` is already included here.
https://github.com/Benjaminimal/PermaplanT/blob/bd1f7916657fb4f3a39196beeab759cf8c5d4091/.gitignore#L275